src/main/kotlin/org/rust/cargo/runconfig/CargoTestRunState.kt

```diff
@@ -13,31 +13,29 @@ import com.intellij.execution.configurations.RunConfiguration
 import com.intellij.execution.runners.ExecutionEnvironment
 import com.intellij.execution.runners.ProgramRunner
 import com.intellij.execution.testframework.autotest.ToggleAutoTestAction
-import com.intellij.execution.testframework.sm.SMTestRunnerConnectionUtil
-import com.intellij.execution.testframework.sm.runner.ui.SMTRunnerConsoleView
 import org.rust.cargo.runconfig.command.CargoCommandConfiguration
-import org.rust.cargo.runconfig.test.CargoTestConsoleProperties
+import org.rust.cargo.runconfig.console.CargoTestConsoleBuilder
 import org.rust.cargo.toolchain.CargoCommandLine
 
 class CargoTestRunState(
     environment: ExecutionEnvironment,
     config: CargoCommandConfiguration.CleanConfiguration.Ok
 ) : CargoRunState(environment, config) {
 
+    init {
+        consoleBuilder = CargoTestConsoleBuilder(environment.runProfile as RunConfiguration, environment.executor)
```

Comment:
Hmm, we creating `CargoConsoleBuilder` inside `CargoRunState` constructor and adding filters to it, and then replacing `consoleBuilder` with `CargoTestConsoleBuilder` (and again adding filters to it). Maybe we can make it a bit clearer, for example create abstract method `creaeConsoleBuilder` and so call it once.